src/lisp/kernel/cmp/compile-file.lisp

```diff
@@ -144,27 +144,20 @@ and the pathname of the source file - this will also be used as the module initi
                (target-options (llvm-sys:make-target-options)))
           (multiple-value-bind (target msg)
               (llvm-sys:target-registry-lookup-target "" triple)
-            (unless target (error msg))
-            (let* ((target-machine (llvm-sys:create-target-machine target
-                                                                   (llvm-sys:get-triple triple)
-                                                                   ""
-                                                                   ""
-                                                                   target-options
-                                                                   reloc-model
-                                                                   (code-model :jit nil :target-faso-file target-faso-file)
-                                                                   'llvm-sys:code-gen-opt-default
-                                                                   NIL ; JIT?
-                                                                   ))
-                   (pm (llvm-sys:make-pass-manager))
-                   (target-pass-config (llvm-sys:create-pass-config target-machine pm))
-                   (_ (llvm-sys:set-enable-tail-merge target-pass-config nil))
-                   (tli (llvm-sys:make-target-library-info-wrapper-pass triple #||LLVM3.7||#))
-                   (data-layout (llvm-sys:create-data-layout target-machine)))
-              (declare (ignore _))
-              (llvm-sys:set-data-layout module data-layout)
-              (llvm-sys:pass-manager-add pm tli)
-              (llvm-sys:add-passes-to-emit-file-and-run-pass-manager target-machine pm output-stream nil #|<-dwo-stream|# file-type module)))))))
-
+            (unless target
+              (error msg))
+            (llvm-sys:emit-module (llvm-sys:create-target-machine target
+                                                                  (llvm-sys:get-triple triple)
+                                                                  ""
+                                                                  ""
+                                                                  target-options
+                                                                  reloc-model
+                                                                  (code-model :jit nil :target-faso-file target-faso-file)
+                                                                  'llvm-sys:code-gen-opt-default
+                                                                  nil)
+                                  output-stream
+                                  nil #|<-dwo-stream|#
```

Comment:
what is dwo stream? Doesn't seem to be documented in LLVM either from a quick search, but might be good to mention... sssssomewhere.

Reply:
Apparently it is a Dwarf Object stream. Improved the comment.